File: LanguAI.Backend/LanguAI.Backend/ViewModels/Exercise/ExerciseViewModel.cs
using LanguAI.Backend.Core.Enums;
using LanguAI.Backend.ViewModels.Card;

namespace LanguAI.Backend.ViewModels.Exercise;

public class ExerciseViewModel
{
    public int ExerciseId { get; set; }

    public string MainSentence { get; set; }

    public ExerciseTypeEnum ExerciseType { get; set; }

    public List<IsCorrectAndTextSentenceViewModel> IsCorrectAndTextSentences { get; set; }

    public bool IsActive { get; set; }

    public List<SentenceAssemblyExerciseWord> SentenceAssemblyExerciseSentence { get; set; }

    public string FirstPartOfTheSentence { get; set; }

    public string LastPartOfTheSentence { get; set; }

    public List<string> Words { get; set; }

    public string CorrectWord { get; set; }
    
    public List<CardViewModel> WordPairingExercise { get; set; }
}

File: LanguAI.Frontend/src/api/models/exercise-view-model.ts
/* tslint:disable */
/* eslint-disable */
import { CardViewModel } from '../models/card-view-model';
import { ExerciseTypeEnum } from '../models/exercise-type-enum';
import { IsCorrectAndTextSentenceViewModel } from '../models/is-correct-and-text-sentence-view-model';
import { SentenceAssemblyExerciseWord } from '../models/sentence-assembly-exercise-word';
export interface ExerciseViewModel {
  correctWord?: string | null;
  exerciseId?: number;
  exerciseType?: ExerciseTypeEnum;
  firstPartOfTheSentence?: string | null;
  isActive?: boolean;
  isCorrectAndTextSentences?: Array<IsCorrectAndTextSentenceViewModel> | null;
  lastPartOfTheSentence?: string | null;
  mainSentence?: string | null;
  sentenceAssemblyExerciseSentence?: Array<SentenceAssemblyExerciseWord> | null;
  wordPairingExercise?: Array<CardViewModel> | null;
  words?: Array<string> | null;
}
